utilities/openshift_resources/cluster_scoped_resource.py

```
@@ -0,0 +1,300 @@
+        try:
+            self._get_json()
+            return True
+        except OCError as err:
+            if "NotFound" in err.stderr or "not found" in err.stderr:
+                return False
+            raise
+
+    @property
+    def instance(self) -> ResourceDict:
+        """Fetch the live resource from the cluster and cache it."""
+        self._instance = ResourceDict(self._get_json())  # noqa: FCN001 - dict subclass constructor
```

Ruff (0.16.0)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)